diff --git a/libcxx/include/__config b/libcxx/include/__config
index 97e33f315..0567b611a 100644
--- a/libcxx/include/__config
+++ b/libcxx/include/__config
@@ -939,8 +939,8 @@ typedef unsigned int   char32_t;
 #   define _LIBCPP_EXTERN_TEMPLATE_EVEN_IN_DEBUG_MODE(...) extern template __VA_ARGS__;
 #endif
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(_LIBCPP_MSVCRT_LIKE) || \
-    defined(__sun__) || defined(__NetBSD__) || defined(__CloudABI__)
+#if (defined(__APPLE__) || defined(__FreeBSD__) || defined(_LIBCPP_MSVCRT_LIKE) || \
+    defined(__sun__) || defined(__NetBSD__) || defined(__CloudABI__)) && !defined(__ORBIS__)
 #define _LIBCPP_LOCALE__L_EXTENSIONS 1
 #endif
 
@@ -976,7 +976,7 @@ typedef unsigned int   char32_t;
 #define _LIBCPP_HAS_DEFAULTRUNELOCALE
 #endif
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__sun__)
+#if (defined(__APPLE__) || defined(__FreeBSD__) || defined(__sun__)) && !defined(__ORBIS__)
 #define _LIBCPP_WCTYPE_IS_MASK
 #endif
 
diff --git a/libcxx/include/locale b/libcxx/include/locale
index 8e584005d..d6aa0ee2a 100644
--- a/libcxx/include/locale
+++ b/libcxx/include/locale
@@ -206,7 +206,7 @@ template <class charT> class messages_byname;
 
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
 // Most unix variants have catopen.  These are the specific ones that don't.
-#  if !defined(__BIONIC__) && !defined(_NEWLIB_VERSION)
+#  if !defined(__BIONIC__) && !defined(_NEWLIB_VERSION) && !defined(__ORBIS__)
 #    define _LIBCPP_HAS_CATOPEN 1
 #    include <nl_types.h>
 #  endif